### ISSUE TITLE
SQLfmt not found but it is installed

### ISSUE THREAD

**Issue by brandon-segal** [innoverio/vscode-dbt-power-user#234]
### Expected behavior

I expect the format on save behavior to work with my power user plug-in when working with jinja SQL files. 

### Actual behavior

![Screenshot 2022-11-29 at 10 44 47 AM](https://user-images.githubusercontent.com/44269822/204575579-ff763e60-d4dd-40a2-ad67-42d9a6fc5852.png)

When I run the same node command that checks for sqlfmt in the node repl in my terminal using the same python environment as the power-user plugin I get the correct location. 

```node
await which.default('sqlfmt')
>> '/Users/me/.pyenv/shims/sqlfmt'
```

### Steps To Reproduce

```
pyenv install 3.8.10
pyenv virtualenv 3.8.10 dbt 
pyenv local dbt 
pyenv activate dbt 
pip install dbt-bigquery==1.2.0
pip install "shandy-sqlfmt[jinjasql]"
```
Then select the dbt virtual env for the python environment and try to save a file 

### Log output/Screenshots

![Screenshot 2022-11-29 at 10 44 47 AM](https://user-images.githubusercontent.com/44269822/204575579-ff763e60-d4dd-40a2-ad67-42d9a6fc5852.png)

### Operating System

13.0.1 (22A400)

### dbt version

1.2.0

### dbt Adapter

bigquery

### dbt Power User version

v0.10.1

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by mdesmet** [comment]
Currently we only support a global installation (available on $PATH).

I see that you are willing to submit a PR. Please contact me if you need help implementing this!

**Comment by marcNY** [comment]
I am having the same issue even if I have install the package on my base installation. Here is the message i am getting: `Could not run sqlfmt. Did you install sqlfmt? Detailed error: Error: Illegal value for line`

**Comment by mdesmet** [comment]
> I am having the same issue even if I have install the package on my base installation. Here is the message i am getting: Could not run sqlfmt. Did you install sqlfmt? Detailed error: Error: Illegal value for line

@jlpjavi : as discussed, please provide a pr for this specific issue

**Comment by medina325** [comment]
It simply does not work for me. Followed the instructions step-by-step, and still get this error message:

![image](https://user-images.githubusercontent.com/56964034/217844758-2c1ab46b-c76a-400a-b956-5a3b4008c7b4.png)

Tried conda and .venv and nothing.

**Comment by ezk84** [comment]
Had trouble getting this to work. Installed on Windows 10 using `pipx` which put the executable in `~/.local/bin`. Even after I added that to my PATH, I kept getting the above error (`Illegal value for 'line'`). I could run `sqlfmt` from my command line.

I solved it by moving the `sqlfmt.exe` file into another directory that is also on my path (`~/bin`, in my case). Weird!

**Comment by pebabion** [comment]
> Had trouble getting this to work. Installed on Windows 10 using `pipx` which put the executable in `~/.local/bin`. Even after I added that to my PATH, I kept getting the above error (`Illegal value for 'line'`). I could run `sqlfmt` from my command line.
> 
> I solved it by moving the `sqlfmt.exe` file into another directory that is also on my path (`~/bin`, in my case). Weird!

I'm having the same issue on MacOS.

**Comment by ranjitparva** [comment]
Same issue on Mac

**Comment by pebabion** [comment]
For anyone who is using VS Code, sqlfmt's documentation [shows how you can do it](https://docs.sqlfmt.com/integrations/vs-code) without dbt-power-user extension.

1. First, install [Run on Save](https://marketplace.visualstudio.com/items?itemName=emeraldwalk.RunOnSave)
2. Configure "Run on Save" to run sqlfmt

**Comment by icemtel** [comment]
Hi, @mdesmet. 

> Currently we only support a global installation (available on $PATH).

If I am use windows, do I need to add it to Windows $PATH? I tried adding to $PATH in my .bashrc file (I am using git bash), so now it works in terminal, but the VS extension does not. 

In case somebody else has the same issue, I found my executable file under "C:\Users\[USER_NAME]\AppData\Roaming\Python\Python310\Scripts"

**Comment by b-ned** [comment]
@ezk84,  @pebabion,
The `illegal value for 'line'` bug is happening after dbt Power User has already identified `sqlfmt`, and has to do with the length of the formatted document being longer than the original document.
This has been solved in #387

@mdesmet,@jlpjavi looking at the number of issues with finding the `sqlfmt` does it make sense to add a setting field where users can define the path to their installation of `sqlfmt`? 
So in the `dbtDocumentFormattingEditProvider.ts` first look for the sqlfmt path defined in settings, if empty check ` $PATH`

**Comment by mdesmet** [comment]
Resolved by https://github.com/innoverio/vscode-dbt-power-user/commit/b21ea11c3d32b4d94bd4e45e4455bc2ef7b0ee2c
